app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java

```diff
@@ -25,4 +42,74 @@ public void onBackPressed() {
     public void onLogOutPressed() {
         view.onLogOutPressed();
     }
+
+    @Override
+    public void checkLocationPermission() {
+        if (!view.locationPermissionGranted()) {
+            view.requestPermissions();
+        } else {
+            setLocationObject();
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    @Override
+    public void setLocationObject() {
+        model.setLocationObject(locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER));
+    }
+
+    public void init() {
+        locationManager = view.getLocationManager();
+        checkLocationPermission();
+        call = model.getWeatherDataFromService(model.getLatitud(), model.getLongitude(),
+                BuildConfig.weather_api_key, ConstantUtils.WEATHER_UNITS);
+        call.enqueue(new Callback<WeatherPojo>() {
+            @Override
+            public void onResponse(Call<WeatherPojo> call, Response<WeatherPojo> response) {
+                view.onWeatherDataGet();
+                weatherData = response.body();
+                view.setCityView(weatherData.getName());
+                view.setTemperatureView(weatherData.getMain().getTemp());
+                setWeatherIcon(weatherData.getWeather().get(ConstantUtils.ZERO).getIcon());
+                view.setWeatherMainView(weatherData.getWeather().get(ConstantUtils.ZERO).getMain());
+            }
+
+            @Override
+            public void onFailure(Call<WeatherPojo> call, Throwable t) {
+                view.fetchingError();
+            }
+        });
+    }
+
+    @Override
+    public void onRequestPermissionResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
```

Comment:
Unnecesary switch case, please merge with the if below

Reply:
Done